### PR TITLE
Add Snapchat passport strategy

### PR DESCRIPTION
**Pull request by davidozhang**
Screen Shot 2020-01-21 at 19 36 17

### CODE REVIEW

**Comment by davidozhang** [comment]
@misspia create a `.env` file with the following keys:
`SNAPCHAT_CLIENT_ID`, `SNAPCHAT_CLIENT_SECRET`

get the values from https://kit.snapchat.com/portal/apps/82004a35-5058-4891-99f3-1ef034edcca7

File: auth.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const router = express.Router();
+
+const snapchatStrategy = require('./strategies/snapchat');
+
+function successHandler(req, res) {
+    console.log('Successful authentication');
+};
+
+function errorHandler(er, req, res, next) {
+    console.log('Failed to authenticate');
```
Comment (misspia) [inline auth.js]: if you [throw](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw) an error, the user will be able to `catch` it. ie
```js
// server
function errorHandler(er, req, res, next) {
  throw new Error('Failed to authenticate');
}

// client 
tryingToAuthenticate()
  .then(res => {})
  .catch(er => {
     console.debug('Yay I caught it')
  });
```

File: strategies/snapchat.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const SnapchatStrategy = require('passport-snapchat').Strategy;
+
+module.exports = new SnapchatStrategy({
+    clientID: process.env.SNAPCHAT_CLIENT_ID,
+    clientSecret: process.env.SNAPCHAT_CLIENT_SECRET,
+    callbackURL: "http://localhost:3000/auth/snapchat/callback",
```
Comment (misspia) [inline strategies/snapchat.js]: nit: single quotes 🙈

File: strategies/snapchat.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    profileFields: ['id', 'displayName', 'bitmoji']
+  },
+  async (accessToken, refreshToken, profile, done) => {
+    const tokens = {
+      accessToken: accessToken,
```
Comment (misspia) [inline strategies/snapchat.js]: node runs with ES6 (javascript with some syntax sugar), so you can actually declare the object like this
```
const tokens = {
  accessToken,
  refreshToken,
};
```

[Reference](https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand) to the shorthand. Basically your variable name becomes the name on the object's key. So if the variable name happens to be the same as what you want to name the key then you can declare it like this

**Review by misspia — COMMENTED** [review]
👌 👌 added some tips

File: auth.js (hunk 1 above)
Comment (misspia) [reply auth.js]: seems like express also has another error catching method `next(err)` for errors from an async function ([ref](http://expressjs.com/en/guide/error-handling.html))

File: index.js [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
 
+app.use(auth);
+
 app.get('/', (req, res) => {
   res.send('ServerWar is live');
 });
 
 app.listen(PORT, () => {
-  console.log(`App is listening on port ${PORT}`);
+  console.log(`ServerWar is listening on port ${PORT}`);
```
Comment (misspia) [inline index.js]: ? 😏 
```suggestion
  console.log(`ServerWar is listening on port ${PORT} 👻`);
```

Comment (davidozhang) [reply index.js]: hehe yas!

File: strategies/snapchat.js (hunk 3 above)
Comment (davidozhang) [reply strategies/snapchat.js]: cool, good to know!

File: auth.js (hunk 1 above)
Comment (davidozhang) [reply auth.js]: cool

File: index.js (hunk 4 above)
Comment (davidozhang) [reply index.js]: Screen Shot 2020-01-21 at 19 36 17

**Comment by misspia** [comment]
merge it :D

**Pull request merged**